unit_tests/customizer/grasp_customization.cpp

```diff
@@ -0,0 +1,150 @@
+        edges.push_back(Edge{m.target, m.start, m.weight, false, true});
+    }
+    std::sort(edges.begin(), edges.end());
+    return partition::MultiLevelGraph<EdgeData, osrm::storage::Ownership::Container>(
+        mlp, max_id + 1, edges);
+}
+}
+
+BOOST_AUTO_TEST_SUITE(grasp_customization_tests)
+
+BOOST_AUTO_TEST_CASE(paper_test)
+{
+    /*
+        Example graph from paper:
+        Note: The paper marks 5 as border node, without having a border edge on level 1.
+              This is wrong hence I inserted an edge from 5 to 7 to make it true.
+        10
+          \
+           11---5---0---3---8---13
+           |   / \ /        |
+           6------1----     |
+           | /         \    |
+           7--------2---4---9---14
+                    |
+                   12
+    */
+    // node:                0  1  2  3  4  5  6  7  8  9 10 11 12 13 14
+    std::vector<CellID> l1{{0, 0, 1, 3, 2, 0, 0, 1, 3, 2, 0, 0, 1, 3, 2}};
+    std::vector<CellID> l2{{0, 0, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 1}};
+    std::vector<CellID> l3{{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}};
+    MultiLevelPartition mlp{{l1, l2, l3}, {4, 2, 1}};
+
+    std::vector<MockEdge> edges = {{0, 1, 1},
+                                   {0, 3, 1},
+                                   {0, 5, 1},
+                                   {1, 4, 1},
+                                   {1, 5, 1},
+                                   {1, 6, 1},
+                                   {2, 4, 1},
+                                   {2, 7, 1},
+                                   {2, 12, 1},
+                                   {3, 8, 1},
+                                   {4, 9, 1},
+                                   {5, 7, 1},
+                                   {5, 11, 1},
+                                   {6, 7, 1},
+                                   {6, 11, 1},
+                                   {8, 9, 1},
+                                   {8, 13, 1},
+                                   {9, 14, 1},
+                                   {10, 11, 1}};
+    auto graph = makeGraph(mlp, edges);
+
+    CellStorage cell_storage(mlp, graph);
+    GRASPStorage grasp_storage(mlp, graph, cell_storage);
+    GRASPCellCustomizer customizer(mlp);
+
+    customizer.Customize(graph, cell_storage, grasp_storage);
+
+    std::vector<MockEdge> down_edges;
+    for (auto node : util::irange<NodeID>(0, graph.GetNumberOfNodes()))
+    {
+        for (auto edge : grasp_storage.GetDownwardEdgeRange(node))
+        {
+            auto source = grasp_storage.GetSource(edge);
+            auto weight = grasp_storage.GetEdgeData(edge).weight;
+            down_edges.push_back(MockEdge{source, node, weight});
+        }
+    }
+    std::sort(down_edges.begin(), down_edges.end());
+
+    std::vector<MockEdge> down_edges_reference = {{0, 5, 1},
+                                                  {0, 6, 2},
+                                                  {0, 7, 3},
```

Comment:
I  think here must be 2 instead of 3, because if edge `5->7` is added the path `0->5->7` is shorter than `0->1->6->7` in paper.